tests-trigger

```diff
@@ -77,14 +77,15 @@ def trigger_pull(api, opts):
         # Keep the old link for reference, until testing starts again
         link = status.get("target_url", None)
         if link:
+            assert isinstance(link, str)
```

Comment:
Without this:

> tests-trigger:80: error: Incompatible types in assignment (expression has type "Mapping[str, JsonValue] | Sequence[JsonValue] | float | bool", target has type "str")  [assignment]

This feels a bit strange to me -- is there a `| str` missing there?

Reply:
I don't see that warning.  I only get:

> tests-trigger:19: error: Call to untyped function "get" in typed context  [no-untyped-call]
> tests-trigger:38: error: Call to untyped function "statuses" in typed context  [no-untyped-call]
> tests-trigger:83: error: Call to untyped function "post" in typed context  [no-untyped-call]

(and you'll need to wait on #6265 for those...)

Reply:
Ah.  I fetched your branch approximately 30 seconds too soon.  Now I understand :)

Reply:
The reason for this is pretty simple, and it lies with the line above:

```python
changes = {"state": "pending", "description": github.NOT_TESTED_DIRECT, "context": context}
```
in that line, all the attributes that you add to the dictionary happen to have string values, so mypy deduces a type of `dict[str, str]`.  You can add `changes: dict[str, JsonValue] =` to help it.

Reply:
Another out would be to replace `link = status.get("target_url", None)` with `link = get_str(status, "target_url", None)`

Reply:
...and yet another out would be for us to define a TypedDict subclass for status objects... but that's a whole other can of worms...

Reply:
actually, I don't like that one.  There's a bunch of other junk that GitHub returns in that object that we shouldn't be copying into the new one...

Reply:
The issue is that `status` came from fetching a JSON blob from GitHub.  The API enforces that it's an object, but doesn't do anything else.  That means that if you `.get()` something out of it, it could be any possible JSON value — not just a string, and that's what mypy is complaining about here.  That's why the assert (or `get_str()`) helps.